### PR TITLE
[PHPUnit] Create AbstractPHPUnitRector

### PR DESCRIPTION
**Pull request by carusogabriel**
This PR solves some problems with PHPUnit Rectors:
- Some assertions weren't being refactored if they extended a `BaseTestCase` class, not PHPUnit's;
- Unify how we candidate our classes to look up for assertions: end with `Test`, as it [is the standard in PHPUnit](https://phpunit.de/getting-started-with-phpunit.html). In the feature, we can remove this check, looking up just for the assertions, but only when improving the speed of Rector.

After that, a bug arises calling `toString` in FuncCall that are actually variables: `$method($argument)`, also fixed with this PR.

### CODE REVIEW

File: src/Rector/AbstractPHPUnitRector.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+use PhpParser\Node;
+use Rector\Node\Attribute;
+
+abstract class AbstractPHPUnitRector extends AbstractRector
+{
+    protected function isInTestClass(Node $node): bool
+    {
+        $className = $node->getAttribute(Attribute::CLASS_NAME);
+
+        return Strings::endsWith((string) $className, 'Test');
```
Comment (TomasVotruba) [inline src/Rector/AbstractPHPUnitRector.php]: Could you add comment there, that this is only approximate assumption?
It might be important in some edge cases, where these Rectorswould fail, so it's easier to debug.

**Comment by TomasVotruba** [comment]
Like it overall!
 
Have you tested that on real project? Does it pass?

File: src/Rector/AbstractPHPUnitRector.php (hunk 1 above)
Comment (TomasVotruba) [reply src/Rector/AbstractPHPUnitRector.php]: Also the note that from PR would be great: https://phpunit.de/getting-started-with-phpunit.html

**Comment by carusogabriel** [comment]
@TomasVotruba `Composer` is a proof of concept of this PR. Run with master branch, and them with this fork, and you'll see the results!

**Comment by TomasVotruba** [comment]
Awesome, thanks :+1:
